sys/net/routing/rpl/Makefile

```diff
@@ -1,16 +1,37 @@
-MODE = $(shell grep "RPL_DEFAULT_MOP RPL_NON_STORING_MODE" ../../include/rpl/rpl_config.h)
-ifneq (,$(MODE))
+# RPL_MOP values by IANA https://tools.ietf.org/html/rfc6550#page-129
+# RPL_MOP_NO_DOWNWARD_ROUTES  0x00
+# RPL_MOP_NON_STORING_MODE    0x01
+# RPL_MOP_STORING_MODE_NO_MC  0x02
+# RPL_MOP_STORING_MODE_MC     0x03
+
+# Set the RPL MOP iff it is matching a provided mode
+ifeq (RPL_MOP_NO_DOWNWARD_ROUTES, $(RPL_MOP))
+	CFLAGS += -DRPL_DEFAULT_MOP=0x00
+endif
+ifeq (RPL_MOP_NON_STORING_MODE, $(RPL_MOP))
+	CFLAGS += -DRPL_DEFAULT_MOP=0x01
 	DIRS += rpl_nonstoring
 else
 	DIRS += rpl_storing
 endif
+ifeq (RPL_MOP_STORING_MODE_NO_MC, $(RPL_MOP))
+	CFLAGS += -DRPL_DEFAULT_MOP=0x02
+endif
+ifeq (RPL_MOP_STORING_MODE_MC, $(RPL_MOP))
+	CFLAGS += -DRPL_DEFAULT_MOP=0x03
+endif
+
+# if no RPL_MOP is matching/provided
+ifeq (,$(findstring $(RPL_MOP), RPL_MOP_NO_DOWNWARD_ROUTES RPL_MOP_NON_STORING_MODE RPL_MOP_STORING_MODE_NO_MC RPL_MOP_STORING_MODE_MC))
```

Comment:
if you use another variable to hold the RPL_DEFAULT_MOP instead of assigning it directly to the CFLAGS, you could check here if this variable is filled or not and set the default value iff not. This way you can get rid of this long string with all modes included. May introduce some problems later if we forget to add new modes in this file.